### PR TITLE
app: use external_ip in ssh command

### PR DESCRIPTION
**Pull request by 0xPoe**

### CODE REVIEW

File: app/pages/index.tsx [hunk 1]
```diff
@@ -151,13 +154,13 @@ const Home: NextPage = () => {
...
-      key: 'ssh_host',
+      dataIndex: 'external_ip',
+      key: 'external_ip',
       render: (_, record: Instance) => {
-        if (!record.ssh_host || !record.ssh_port) {
+        if (!record.external_ip) {
           return <Spin />
         }
-        const sshCommand = `ssh root@${record.ssh_host} -p ${record.ssh_port}`
+        const sshCommand = `ssh root@${record.external_ip} -p`
```
Comment (0xPoe) [inline app/pages/index.tsx]:
```suggestion
        const sshCommand = `ssh root@${record.external_ip} `
```

**Comment by sleepymole** [comment]
How about showing external ip in a column explicitly? Although currently only port 22 can be used. After we patch kube-proxy, it will behave like a real ip for instance. If so, seems the internal ip also need to show, otherwise it looks a little weird.🙃

**Comment by 0xPoe** [comment]
> How about showing external ip in a column explicitly? Although currently only port 22 can be used. After we patch kube-proxy, it will behave like a real ip for instance. If so, seems the internal ip also need to show, otherwise it looks a little weird.🙃

So do we still need to provide the command?

**Comment by sleepymole** [comment]
> > How about showing external ip in a column explicitly? Although currently only port 22 can be used. After we patch kube-proxy, it will behave like a real ip for instance. If so, seems the internal ip also need to show, otherwise it looks a little weird.upside_down_face
> 
> So do we still need to provide the command?

Retain command but not show command content?

image

This is the GCP dashboard. It's very similar to our page.

**Comment by 0xPoe** [comment]
> Retain command but not show command content?

It is hidden because we can connect directly in the browser, right? I think we'd better show it, but the table does have too many columns now.
Maybe we can merge a few columns of the resource? I'm not sure if that's a good idea.

**Comment by 0xPoe** [comment]
Also is the internal ip and host a bit redundant?

**Comment by sleepymole** [comment]
> It is hidden because we can connect directly in the browser, right?

Yes. 

> the table does have too many columns now.

This is why I want to hide the content.

> Maybe we can merge a few columns of the resource

Agree.  Another way to add a horizontal scrollbar. But too many columns seems not a good idea. 🤣

**Comment by sleepymole** [comment]
> Also is the internal ip and host a bit redundant?

Yes, internal ip is rarely used. If we don't want to list too many columns, it can be ignored. But for host, I would prefer retain it. A domain is more stable and easier to use than an ip address.

**Comment by sleepymole** [comment]
Many cloud providers (e.g. Aliyun, GCP, Vultr) have a detail page for each instance. That may be the ultimate solution to our too many columns issue. We may consider implementing it in the future.

**Pull request closed without merging**